engine/src/conversion/analysis/struct_accessors.rs

```diff
@@ -0,0 +1,142 @@
+                syn::Fields::Named(named) => {
+                    for field in named.named.iter() {
+                        if !should_generate_accessor(field) {
+                            continue;
+                        }
+
+                        let field_name = field.ident.as_ref().unwrap().to_string();
+                        let accessor_name = get_accessor_name(&struct_name.name, &field_name);
+                        let ident = accessor_name.get_final_ident();
+
+                        // Don't generate accessors that would conflict with existing api (i.e., if a method with the name we would generate already exists)
+                        // TOOD: it would be nicer for this to be in should_generate_accessor
+                        if existing_api.contains(&accessor_name) {
+                            continue;
+                        }
+
+                        let body = CppFunctionBody::ReturnFieldAccess(make_ident(field_name));
+
+                        let struct_type = struct_name.name.to_type_path();
+
+                        let fnarg: FnArg = parse_quote! {
+                            this: *const #struct_type
+                        };
+
+                        // TODO: convert arrays to pointers?
+                        let field_type = &field.ty;
+
+                        accessors.push(Api::Function {
+                            name: ApiName::new_from_qualified_name(accessor_name),
+                            fun: Box::new(FuncToConvert {
+                                provenance: Provenance::SynthesizedOther,
+                                ident,
+                                doc_attrs: Vec::new(),
+                                inputs: [fnarg].into_iter().collect(),
+                                variadic: false,
+                                output: parse_quote! {
+                                    -> #field_type
+                                },
+                                vis: parse_quote! { pub },
+                                virtualness: crate::conversion::api::Virtualness::None,
+                                cpp_vis: crate::conversion::api::CppVisibility::Public,
+                                special_member: Some(SpecialMemberKind::GeneratedAccessor),
+                                unused_template_param: false,
+                                references: Default::default(),
+                                original_name: None,
+                                self_ty: None,
+                                synthesized_this_type: None,
+                                add_to_trait: None,
+                                synthetic_cpp: Some((body, CppFunctionKind::Function)),
+                                is_deleted: false,
+                            }),
+                            analysis: (),
+                        })
+                    }
+                }
+                // TODO: is it even possible for these to occur?
```

Comment:
Probably not, but as a matter of policy I try to assume bindgen might give us weird input for reasons I haven't anticipated, so I just ignore such things rather than `todo!()`.